Source/WebKit/Shared/RemoteLayerTree/RemoteLayerTreePropertyApplier.mm

```diff
@@ -299,8 +299,14 @@ static void updateCustomAppearance(CALayer *layer, GraphicsLayer::CustomAppearan
 
 #if HAVE(AVKIT)
     if (properties.changedProperties & LayerChange::VideoGravityChanged) {
-        if ([layer respondsToSelector:@selector(setVideoGravity:)])
-            [(WebAVPlayerLayer*)layer setVideoGravity:convertMediaPlayerToAVLayerVideoGravity(properties.videoGravity)];
+        auto *playerLayer = layer;
+#if PLATFORM(IOS_FAMILY)
+        if (layerTreeNode)
+            playerLayer = layerTreeNode->uiView().subviews.firstObject.layer;
+#endif
```

Comment:
This is very fragile. I suggest you add a check that the `[layerTreeNode->uiView() isKindOfClass:WKVideoView.class]`, add a `@property (retain, nonatomic) WebAVPlayerLayerView* playerLayerView` to WKVideoView (which adds the playerLayerView to it's view hierarchy), and query that property rather than relying on the playerLayer being the first subview's layer.